### ISSUE TITLE
Add support for PHP 8.3

### ISSUE THREAD

**Issue by localheinz** [FakerPHP/Faker#694]
In all likelihood, [PHP 8.3 will be released on November 23, 2023](https://wiki.php.net/todo/php83), and it would be great if we could support PHP 8.3 soon after the official release.

- [x] require tests to pass on PHP 8.3
    - see #693
    - see #844 
- [x] stop using `MT_RAND_PHP` constant deprecated on PHP 8.3
    - see https://github.com/php/php-src/pull/11560
    - see https://wiki.php.net/rfc/deprecations_php_8_3#mt_rand_php
    - see #844

**Comment by ejunker** [comment]
PHP 8.3 was released. What is remaining? The main issue I notice is with the deprecated `MT_RAND_PHP` constant.

Could it be solved by doing something like this?
```php
$mode = PHP_VERSION_ID >= 80300 ? MT_RAND_MT19937 : MT_RAND_PHP;
mt_srand((int) $seed, $mode);
```

**Comment by localheinz** [comment]
Fixed with #844.

**Comment by cs278** [comment]
This bug fix is arguably a BC break, it means Faker's output between differs when switching PHP version when using the same seed.

If we're happy changing the seeding in a patch release why not just use `MT_RAND_MT19937` all the time? That mode is available on all supported PHP versions and means Faker doesn't change behaviour between PHP versions.
